lib/evaluation/handlers/limitDisclosureEvaluationHandler.ts

```diff
@@ -134,15 +135,43 @@ export class LimitDisclosureEvaluationHandler extends AbstractEvaluationHandler
     // Can be nested array / object
     const presentationFrame: SdJwtPresentationFrame = {};
 
+    const processNestedObject = (obj: ClaimValue, currentPath: PathComponent[], basePath: PathComponent[]) => {
+      if (obj === null || typeof obj !== 'object') {
+        // For literal values, set the path to true in the presentation frame
+        JsonPathUtils.setValue(presentationFrame, currentPath, true);
+        return;
+      }
+
+      // For arrays, process each element
+      if (Array.isArray(obj)) {
+        obj.forEach((item, index) => {
+          processNestedObject(item, [...currentPath, index], basePath);
+        });
+        return;
+      }
+
+      // For objects, process each child property
+      Object.entries(obj).forEach(([key, value]) => {
+        processNestedObject(value, [...currentPath, key], basePath);
+      });
+    };
+
     for (const { inputDescriptor, inputDescriptorIndex } of inputDescriptors) {
       for (const field of inputDescriptor.constraints?.fields ?? []) {
         if (field.path) {
           const inputField = JsonPathUtils.extractInputField(vc.decodedPayload, field.path);
 
-          // We set the value to true at the path in the presentation frame,
           if (inputField.length > 0) {
             const selectedField = inputField[0];
-            JsonPathUtils.setValue(presentationFrame, selectedField.path, true);
+            const fieldValue = JsonPathUtils.getValue<ClaimValue>(vc.decodedPayload, selectedField.path);
+
+            if (fieldValue !== null && typeof fieldValue === 'object') {
+              // For objects, recursively process all nested fields
+              processNestedObject(fieldValue, selectedField.path, selectedField.path);
+            } else {
+              // For literal values, just set the path to true
+              JsonPathUtils.setValue(presentationFrame, selectedField.path, true);
+            }
```

Comment:
Curious, why do we need to handle nested objects? If a PD includes `$.address` and address is an object, then having `address: true` in the disclosure frame should be enough right?

Reply:
@TimoGlastra 
This is for the situation where a sd-jwt (not created by us) does not have a disclosure frame element on a group, but on the individual fields within a group. ie.
```
"electronicPassport": {
  "dataGroup1": {
     "birthdate": "{{birthdate}}",
     "expiryDate": "{{{ dateTimeAfterDays 1080 }}}",
     "passportNumberIdentifier": "{{passportNumberIdentifier}}"
  },
  "disclosureFrame": {
    "electronicPassport": {
      "dataGroup1": {
        "_sd": [
          "birthdate",
          "expiryDate",
          "passportNumberIdentifier"
        ]
      },
}
```
When the pd asks to disclose  \$.electronicPassport.dataGroup1 instead of \$.electronicPassport.dataGroup1.* it should still disclose all disclosable fields & groups within the group.
(When you control PD or credential you can make it work by modifying either on or both, but we ran into an interoperability tests where you we do not.)
Also with the address example in the Funke PID, the @sd-jwt library (0.7.2) does not disclose anything when specifying address: true

Reply:
Forking the @sd-jwt library which is not ours is also something we would like to avoid. When I have more time I can play around with the tests in the @sd-jwt library, but have plenty of other stuff on my plate atm.

Reply:
For clarity, for this setup address: true does work:
```
 "disclosureFrame": {
    "electronicPassport": {
        "_sd": [
          "dataGroup1"
        ]
}
```
But then you always disclose the whole address and just field passportNumberIdentifier won't work anymore.